### PR TITLE
Switched to coincurve

### PR DESCRIPTION
**Pull request by cag**
This is a continuation of https://github.com/ethereum/pyethereum/pull/713 which is meant to address https://github.com/ethereum/pyethereum/issues/708

I accidentally deleted the original fork.

@ofek

### CODE REVIEW

**Comment by ofek** [comment]
Fantastic, thanks!

**Comment by ofek** [comment]
Tests fail atm

**Comment by cag** [comment]
Yeah 😫

Or do you mean the ones dealing with signatures? I didn't check because the tests were totally fubar.

File: ethereum/utils.py [hunk 1]
```diff
@@ -13,11 +13,11 @@
 
 
 try:
-    import secp256k1
+    import coincurve
 except ImportError:
-    import warnings
-    warnings.warn('could not import secp256k1', ImportWarning)
-    secp256k1 = None
+    import warning
```
Comment (ofek) [inline ethereum/utils.py]: @cag This should indeed be `warnings`. After you remove this tests may pass.

Comment (ofek) [inline ethereum/utils.py]: @cag Can you please change this?

**Comment by ofek** [comment]
@vbuterin @karlfloersch @joeykrug This looks ready for merge. Can we please get a review? Afterward @cag will add coincurve to requirements.txt.

This fixes #63 #396 #708 #762 #782. supports #188 and #132

**Comment by cag** [comment]
Done. Hi Joey :P

**Comment by ofek** [comment]
@cag Needs a rebase

**Comment by ofek** [comment]
@cag Great, now just add `coincurve>=5.0.1` to requirements.txt

**Comment by ofek** [comment]
@cag Sorry, it seems we need to change `warning` back to `warnings`. Then that's it!!

**Comment by ofek** [comment]
@joeykrug All ready

**Comment by ofek** [comment]
@joeykrug Are we OK to merge?

**Comment by joeykrug** [comment]
Merged

**Comment by joeykrug** [comment]
Won't let me merge on mobile haha, will do in 5 min

**Comment by ofek** [comment]
@joeykrug ok :+1:

**Comment by ofek** [comment]
@joeykrug Thanks!

@cag Well done, and thanks for the time you spent on this!

**Comment by cag** [comment]
Haha, you're welcome

**Pull request merged**